nb2workflow/deploy.py

```diff
@@ -69,24 +85,62 @@ def deploy(git_origin, deployment_base_name, namespace="oda-staging", local=Fals
 ENV ODA_WORKFLOW_VERSION="{descr}"
 ENV ODA_WORKFLOW_LAST_AUTHOR="{author}"
 ENV ODA_WORKFLOW_LAST_CHANGED="{last_change_time}"
+ENV ODA_WORKFLOW_NOTEBOOK_PATH="{notebook_fullpath_in_container}"
 
 COPY nb-repo/ /repo/
 
 RUN curl -o /usr/bin/jq -L https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64; chmod +x /usr/bin/jq
-RUN for nn in /repo/*.ipynb; do mv $nn $nn-tmp;  jq '.metadata.kernelspec.name |= "python3"' $nn-tmp > $nn ; rm $nn-tmp ; done
+RUN for nn in $ODA_WORKFLOW_NOTEBOOK_PATH/*.ipynb; do mv $nn $nn-tmp;  jq '.metadata.kernelspec.name |= "python3"' $nn-tmp > $nn ; rm $nn-tmp ; done
 
-ENTRYPOINT nb2service --debug /repo/ --host 0.0.0.0 --port 8000 | cut -c1-500
+ENTRYPOINT nb2service --debug $ODA_WORKFLOW_NOTEBOOK_PATH --host 0.0.0.0 --port 8000 | cut -c1-500
 """)
 
         image = f"odahub/nb-{pathlib.Path(git_origin).name}:{descr}-nb2w{version()}" # {time.strftime(r'%y%m%d%H%M%S')}"
 
         subprocess.check_call( # cli is more stable than python API
             ["docker", "build", ".", "-t", image],
-            cwd=tmpdir)
+            cwd=tmpdir)        
+
+        if run_tests: 
+            # TODO: run tests too
+            out = subprocess.check_output(
+                    ["docker", "run", '--rm', '--entrypoint', 'bash', image, '-c', 
+                     ('pip install nb2workflow[rdf,mmoda,service] --upgrade;'
+                      'for a in $(ls $ODA_WORKFLOW_NOTEBOOK_PATH/*ipynb | grep -v test_); do'
+                      '  nbinspect --machine-readable $a;'
+                      '  nbrun --machine-readable $a;'
+                      'done')
+                     ],
+                    cwd=tmpdir)
+
+            workflow_dispatcher_signature = json.loads(re.search(rb"^WORKFLOW-DISPATCHER-SIGNATURE: (.*?)$", out, re.M).group(1).decode())
+            workflow_nb_signature = json.loads(re.search(rb"^WORKFLOW-NB-SIGNATURE: (.*?)$", out, re.M).group(1).decode())
+        else:
+            workflow_dispatcher_signature = None
+            workflow_nb_signature = None
+
+        if run_tests: 
```

Comment:
Duplication?

Reply:
actually it belongs to another PR, I remove https://github.com/oda-hub/nb2workflow/pull/31

Reply:
on second thought, since you looked on it already, I combine them both here.